frontend/src/pages/projects/screens/detail/storage/StorageTable.tsx

```diff
@@ -15,20 +19,66 @@ type StorageTableProps = {
 const StorageTable: React.FC<StorageTableProps> = ({ pvcs, refresh, onAddPVC }) => {
   const [deleteStorage, setDeleteStorage] = React.useState<PersistentVolumeClaimKind | undefined>();
   const [editPVC, setEditPVC] = React.useState<PersistentVolumeClaimKind | undefined>();
+  const isStorageClassesAvailable = useIsAreaAvailable(SupportedArea.STORAGE_CLASSES).status;
+  const [alertStatuses, setAlertStatuses] = React.useState<Record<string, boolean>>({});
+
+  const isDeprecatedAlert = React.useMemo(
+    () => Object.values(alertStatuses).includes(true),
+    [alertStatuses],
+  );
+  const [storageClasses, storageClassesLoaded] = useStorageClasses();
+  const storageClassConfigMap = React.useMemo(
+    () =>
+      storageClasses.reduce((acc: Record<string, StorageClassConfig | undefined>, sc) => {
+        acc[sc.metadata.name] = getStorageClassConfig(sc);
+
+        return acc;
+      }, {}),
+    [storageClasses],
+  );
```

Comment:
why use a map of the configs and pass it in. just fetch the storageClass metadata in the row itself

Reply:
The reason to map the storageClass with storageConfig , so we can directly get the storageConfig for the corresponding to `pvc.spec.storageClassName`

Do you want me to do this instead in the `StorageTableRow`??
 ```
const storageClassAvailable = storageClasses.find(
    (sc) => sc.metadata.name === obj.spec.storageClassName,
  );

  const storageClass = storageClassAvailable
    ? getStorageClassConfig(storageClassAvailable)
    : undefined;

```

We only get `storageClassName` which is a `string` from `PersistentVolumeClaimKind`
and we need to iterate over  the storageClasses to find ClassConfig

Reply:
i see. how about you just pass in the storage class config into the row. so instead of passing in the whole map, you just pass in the storage class config like so

```tsx
...
storageClassConfig={storageClassConfigMap[obj.spec.storageClassName ?? '']}
...